dbusmock/templates/systemd-base.py

```diff
@@ -0,0 +1,124 @@
+'''systemd mock template base
+'''
+
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the Free
+# Software Foundation; either version 3 of the License, or (at your option) any
+# later version.  See http://www.gnu.org/copyleft/lgpl.html for the full text
+# of the license.
+
+__author__ = 'Jonas Ådahl'
+__copyright__ = '(c) 2021 Red Hat'
+
+from gi.repository import GLib
+
+from dbusmock import MOCK_IFACE, mockobject
+
+BUS_PREFIX = 'org.freedesktop.systemd1'
+PATH_PREFIX = '/org/freedesktop/systemd1'
+
+
+BUS_NAME = BUS_PREFIX
+MAIN_OBJ = PATH_PREFIX
+MAIN_IFACE = BUS_PREFIX + '.Manager'
+UNIT_IFACE = BUS_PREFIX + '.Unit'
+
+
+def load_base(mock, _parameters):
+    mock.next_job_id = 1
+    mock.units = {}
+
+    mock.AddMethod(MAIN_IFACE,
+                   'StartUnit',
+                   'ss', 'o',
+                   'ret = self.StartUnit(self, *args)')
+    mock.AddMethod(MAIN_IFACE,
+                   'StartTransientUnit',
+                   'ssa(sv)a(sa(sv))', 'o',
+                   'ret = self.StartTransientUnit(self, *args)')
+    mock.AddMethod(MAIN_IFACE,
+                   'StopUnit',
+                   'ss', 'o',
+                   'ret = self.StopUnit(self, *args)')
+    mock.AddMethod(MAIN_IFACE,
+                   'GetUnit',
+                   's', 'o',
+                   'ret = self.GetUnit(self, *args)')
+    mock.AddProperties(MAIN_IFACE, {'Version': 'v246'})
+
+    mock.StartUnit = StartUnit
```

Comment:
Why this indirection? If you annotate `def StartUnit()` with ``@dbus.service.method`, it should be much simpler?

Reply:
The indirection was added due to using the decoration did not work when the implementation was in the `systemd-base.py` but the template itself was `systemd-user.py`. That is the reason for adding the `AddServiceMethod()` API.

As the decorations did not work in this case, but still wanting to use decorations, I added said method, but did *not* expose it via D-Bus (I did not decorate it). Maybe it would be clearer if it used lower case.

Reply:
No longer relevant, so resolving.